File: ClaimAuto.HealthSystems.Server/DTOs/PolicyDTO.cs
namespace ClaimAuto.HealthSystems.Server.DTOs
{
    // ── CreatePolicyDto
    public class CreatePolicyDto
    {
        public string PlanCode { get; set; } = string.Empty;     // e.g. "FAMILY-GOLD-2024"
        public string PlanName { get; set; } = string.Empty;
        public string? CoverageRulesJSON { get; set; }
        public decimal? SumInsured { get; set; }                 // total insurer payout limit per year
        public decimal? DeductibleAmount { get; set; }
        public DateTime EffectiveFrom { get; set; }              // set once — never changed
        public DateTime? EffectiveTo { get; set; }
    }

    // ── UpdatePolicyDto
    public class UpdatePolicyDto
    {
        public string? PlanName { get; set; }
        public string? CoverageRulesJSON { get; set; }
        public decimal? SumInsured { get; set; }                 // total insurer payout limit per year
        public decimal? DeductibleAmount { get; set; }
        public DateTime? EffectiveTo { get; set; }
        public string? Status { get; set; }    // "Active","Expired","Suspended"
    }

    // ── PolicyResponseDto
    public class PolicyResponseDto
    {
        public int PolicyID { get; set; }
        public string PlanCode { get; set; } = string.Empty;
        public string PlanName { get; set; } = string.Empty;
        public string? CoverageRulesJSON { get; set; }
        public decimal? SumInsured { get; set; }                 // total insurer payout limit per year
        public decimal? DeductibleAmount { get; set; }
        public DateTime EffectiveFrom { get; set; }
        public DateTime? EffectiveTo { get; set; }
        public string Status { get; set; } = string.Empty;
        public int MemberCount { get; set; }                     // computed — not in DB model
    }
}

File: claimauto.healthsystems.client/src/models/policies/PolicyDto.js
// ── CREATE POLICY ─────────────────────────────────────────────────────────────
// Used when: Admin fills in the "New Policy" form and clicks Create
// Matches:   POST /api/policies  →  backend CreatePolicyDto.cs

export class CreatePolicyDto {
  constructor({
    planCode          = '',   // e.g. "FAMILY-GOLD-2025"  — unique, locked after save
    planName          = '',   // e.g. "Family Gold Health Plan"
    coverageRulesJSON = '',   // optional JSON string
    sumInsured        = '',   // total insurer payout limit per policy year
    deductibleAmount  = '',   // number — e.g. 5000
    effectiveFrom     = '',   // date string — e.g. "2025-01-01"
    effectiveTo       = '',   // date string — blank means auto-renewing
  } = {}) {
    this.planCode          = planCode;
    this.planName          = planName;
    this.coverageRulesJSON = coverageRulesJSON;
    this.sumInsured        = sumInsured;
    this.deductibleAmount  = deductibleAmount;
    this.effectiveFrom     = effectiveFrom;
    this.effectiveTo       = effectiveTo;
  }
}


// ── UPDATE POLICY ─────────────────────────────────────────────────────────────
// Used when: Admin edits an existing policy
// Matches:   PUT /api/policies/{id}  →  backend UpdatePolicyDto.cs
// NOTE:      planCode and effectiveFrom are NOT here — they're locked after creation

export class UpdatePolicyDto {
  constructor({
    planName          = '',
    coverageRulesJSON = '',
    sumInsured        = '',   // total insurer payout limit per policy year
    deductibleAmount  = '',
    effectiveTo       = '',
    status            = '',   // "Active", "Expired", "Suspended"
  } = {}) {
    this.planName          = planName;
    this.coverageRulesJSON = coverageRulesJSON;
    this.sumInsured        = sumInsured;
    this.deductibleAmount  = deductibleAmount;
    this.effectiveTo       = effectiveTo;
    this.status            = status;
  }
}